src/main/java/io/airlift/slice/SliceUtf8.java

```diff
@@ -37,28 +39,45 @@ private SliceUtf8() {}
 
     private static final int[] LOWER_CODE_POINTS;
     private static final int[] UPPER_CODE_POINTS;
+    private static final boolean[] PUNCTUATION_CODE_POINTS;
     private static final boolean[] WHITESPACE_CODE_POINTS;
 
     static {
         LOWER_CODE_POINTS = new int[MAX_CODE_POINT + 1];
         UPPER_CODE_POINTS = new int[MAX_CODE_POINT + 1];
         WHITESPACE_CODE_POINTS = new boolean[MAX_CODE_POINT + 1];
+        PUNCTUATION_CODE_POINTS = new boolean[MAX_CODE_POINT + 1];
         for (int codePoint = 0; codePoint <= MAX_CODE_POINT; codePoint++) {
             int type = Character.getType(codePoint);
             if (type != Character.SURROGATE) {
                 LOWER_CODE_POINTS[codePoint] = Character.toLowerCase(codePoint);
                 UPPER_CODE_POINTS[codePoint] = Character.toUpperCase(codePoint);
                 WHITESPACE_CODE_POINTS[codePoint] = Character.isWhitespace(codePoint);
+                PUNCTUATION_CODE_POINTS[codePoint] = isPunctuation(type);
             }
             else {
                 LOWER_CODE_POINTS[codePoint] = REPLACEMENT_CODE_POINT;
                 UPPER_CODE_POINTS[codePoint] = REPLACEMENT_CODE_POINT;
                 WHITESPACE_CODE_POINTS[codePoint] = false;
+                PUNCTUATION_CODE_POINTS[codePoint] = false;
             }
         }
     }
 
     /**
+     * Utility method to identify if the character of a type is a punctuation character
+     * @param type - type of the character
+     * @return true if the character of type is a punctuation character (excluding quote and continuation
+     *  punctuation)
+     */
+    private static boolean isPunctuation(int type) {
```

Comment:
This is a bad name because 3 out of 7 `P` categories (`Pc`, `Pf`, `Pi`) are excluded.